### PR TITLE
[refactor] Grants: Wrap grants with their valid privilges for cztack codegen

### PR DESCRIPTION
**Pull request by edulop91**
Wrapping some grant resources with more context so that it can be consumed for cogegen terraform modules in cztack. In particular, trying to automate the generate of the ALL grant modules for all types the provider supports. 

See https://github.com/chanzuckerberg/cztack/pull/262 for more context

## Test Plan

* [ ] acceptance tests

* [ ] …

## References

*

### CODE REVIEW

**Review by ryanking — APPROVED** [review]

File: pkg/resources/helpers_test.go [hunk 1]
```diff
@@ -120,6 +120,14 @@ func storageIntegration(t *testing.T, id string, params map[string]interface{})
 	return d
 }
 
+func stream(t *testing.T, id string, params map[string]interface{}) *schema.ResourceData {
```
Comment (cziie) [inline pkg/resources/helpers_test.go:123]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>stream redeclared in this block

File: pkg/resources/stream_acceptance_test.go [hunk 2]
```diff
@@ -8,7 +8,7 @@ import (
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 )
 
-func TestAcc_Stream(t *testing.T) {
+func TestAccStream(t *testing.T) {
```
Comment (cziie) [inline pkg/resources/stream_acceptance_test.go:11]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>AT003: acceptance test function name should include underscore

File: pkg/resources/stream.go [hunk 3]
```diff
@@ -243,17 +243,20 @@ func DeleteStream(d *schema.ResourceData, meta interface{}) error {
...
 
 	return nil
 }
 
 // UpdateStream implements schema.UpdateFunc
-func UpdateStream(d *schema.ResourceData, meta interface{}) error {
-	streamID, err := streamIDFromString(d.Id())
+func UpdateStream(data *schema.ResourceData, meta interface{}) error {
+	// https://www.terraform.io/docs/extend/writing-custom-providers.html#error-handling-amp-partial-state
+	data.Partial(true)
```
Comment (cziie) [inline pkg/resources/stream.go:257]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>R007: deprecated data.Partial(true)

File: pkg/resources/stream.go [hunk 4]
```diff
@@ -265,14 +268,14 @@ func UpdateStream(d *schema.ResourceData, meta interface{}) error {
 	builder := snowflake.Stream(streamName, dbName, schema)
 
 	db := meta.(*sql.DB)
-	if d.HasChange("comment") {
-		comment := d.Get("comment")
+	if data.HasChange("comment") {
+		_, comment := data.GetChange("comment")
```
Comment (cziie) [inline pkg/resources/stream.go:272]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>R010: prefer d.Get() over d.GetChange() when only using second return value

File: pkg/resources/stream.go [hunk 5]
```diff
@@ -152,13 +152,13 @@ func streamOnTableIDFromString(stringID string) (*streamOnTableID, error) {
 }
 
 // CreateStream implements schema.CreateFunc
-func CreateStream(d *schema.ResourceData, meta interface{}) error {
+func CreateStream(data *schema.ResourceData, meta interface{}) error {
```
Comment (cziie) [inline pkg/resources/stream.go:155]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>R014: *schema.ResourceData parameter of CreateFunc, ReadFunc, UpdateFunc, or DeleteFunc should be named d

File: pkg/resources/stream.go [hunk 6]
```diff
@@ -190,15 +190,15 @@ func CreateStream(d *schema.ResourceData, meta interface{}) error {
...
-	d.SetId(dataIDInput)
+	data.SetId(dataIDInput)
 
-	return ReadStream(d, meta)
+	return ReadStream(data, meta)
 }
 
 // ReadStream implements schema.ReadFunc
-func ReadStream(d *schema.ResourceData, meta interface{}) error {
+func ReadStream(data *schema.ResourceData, meta interface{}) error {
```
Comment (cziie) [inline pkg/resources/stream.go:199]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>R014: *schema.ResourceData parameter of CreateFunc, ReadFunc, UpdateFunc, or DeleteFunc should be named d

File: pkg/resources/stream.go [hunk 7]
```diff
@@ -228,9 +228,9 @@ func ReadStream(d *schema.ResourceData, meta interface{}) error {
 }
 
 // DeleteStream implements schema.DeleteFunc
-func DeleteStream(d *schema.ResourceData, meta interface{}) error {
+func DeleteStream(data *schema.ResourceData, meta interface{}) error {
```
Comment (cziie) [inline pkg/resources/stream.go:231]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>R014: *schema.ResourceData parameter of CreateFunc, ReadFunc, UpdateFunc, or DeleteFunc should be named d

File: pkg/resources/stream.go [hunk 8]
```diff
@@ -243,17 +243,20 @@ func DeleteStream(d *schema.ResourceData, meta interface{}) error {
...
-	d.SetId("")
+	data.SetId("")
 
 	return nil
 }
 
 // UpdateStream implements schema.UpdateFunc
-func UpdateStream(d *schema.ResourceData, meta interface{}) error {
-	streamID, err := streamIDFromString(d.Id())
+func UpdateStream(data *schema.ResourceData, meta interface{}) error {
```
Comment (cziie) [inline pkg/resources/stream.go:255]: **[tfproviderlint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>R014: *schema.ResourceData parameter of CreateFunc, ReadFunc, UpdateFunc, or DeleteFunc should be named d

File: pkg/resources/stream.go [hunk 9]
```diff
@@ -68,7 +68,7 @@ func Stream() *schema.Resource {
 
 		Schema: streamSchema,
 		Importer: &schema.ResourceImporter{
-			StateContext: schema.ImportStatePassthroughContext,
+			State: schema.ImportStatePassthrough,
```
Comment (cziie) [inline pkg/resources/stream.go:71]: 🚫 **[golangci]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>SA1019: schema.ImportStatePassthrough is deprecated: Please use the context aware ImportStatePassthroughContext instead  (staticcheck)

**Pull request merged**